extensions/ql-vscode/src/pure/errors.ts

```diff
@@ -54,19 +54,34 @@ export function redactableError(
   ...values: unknown[]
 ): RedactableError;
 export function redactableError(
-  error: Error,
+  error: ErrorLike,
 ): (strings: TemplateStringsArray, ...values: unknown[]) => RedactableError;
 
 export function redactableError(
-  errorOrStrings: Error | TemplateStringsArray,
+  errorOrStrings: ErrorLike | TemplateStringsArray,
   ...values: unknown[]
 ):
   | ((strings: TemplateStringsArray, ...values: unknown[]) => RedactableError)
   | RedactableError {
-  if (errorOrStrings instanceof Error) {
+  if (isErrorLike(errorOrStrings)) {
     return (strings: TemplateStringsArray, ...values: unknown[]) =>
       new RedactableError(errorOrStrings, strings, values);
   } else {
     return new RedactableError(undefined, errorOrStrings, values);
   }
 }
+
+export interface ErrorLike {
+  message: string;
+  stack?: string;
+}
+
+function isErrorLike(error: any): error is ErrorLike {
+  if (
+    typeof error.message === "string" &&
+    (typeof error.stack === "undefined" || typeof error.stack === "string")
```

Comment:
I'm not sure what the `typeof` for `undefined` adds here. Would this be equivalent?

```suggestion
    (error.stack === undefined || typeof error.stack === "string")
```

Reply:
I don't think it adds anything except then all the checks are "typeof". But I agree just checking directly against `undefined` is equivalent and more understandable. I added this earlier because of a suggestion from Andrew but I think I may have misunderstood what he was saying.

I'm pretty confident what you're suggesting here is correct and it's what I had originally in this PR, so I'll just go with this.